source/adapters/level_zero/v2/event_pool_cache.cpp

```diff
@@ -16,29 +16,32 @@ namespace v2 {
 event_pool_cache::event_pool_cache(size_t max_devices,
                                    ProviderCreateFunc ProviderCreate)
     : providerCreate(ProviderCreate) {
-  pools.resize(max_devices);
+  pools.resize(max_devices * (1ULL << EVENT_FLAGS_USED_BITS));
```

Comment:
You could define `MAX_EVENT_FLAGS = UR_BIT(3),` and use it here.

Reply:
I could, but I still need EVENT_FLAGS_USED_BITS to know how much do I need to shift device Id when creating index for the pool vector. I think it's better to just have a single variable for that.